spec/lib/booking_requests/booking_requests_spec.rb

```diff
@@ -2,8 +2,16 @@
   let(:api) { instance_double(BookingRequests::Api) }
 
   describe '.slots' do
-    before do
+    around do |example|
+      previous_api = BookingRequests.api
       BookingRequests.api = api
+
+      example.run
+
+      BookingRequests.api = previous_api
```

Comment:
Good plan on both counts, will take a look.

Side note: pretty sure we're employing this pattern in a few places so we quite possibly should go search out and refactor elsewhere too.